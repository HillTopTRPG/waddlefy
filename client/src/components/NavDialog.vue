<template>
  <v-dialog
    :model-value="modelValue"
    :contained="true"
    :fullscreen="true"
    :scrim="false"
    :no-click-animation="true"
    :persistent="true"
    class="my-16 nav-dialog"
    transition="slide-y-reverse-transition"
  >
    <v-card>
      <v-card-title class="d-flex justify-space-between align-center py-1">
        <span data-cy="title-text">{{ title }}</span>
        <v-btn icon="mdi-close" variant="text" size="small" style="font-size: 0.8em" @click="emits('close')"></v-btn>
      </v-card-title>
      <slot />
    </v-card>
  </v-dialog>
</template>

<script lang="ts" setup>
defineProps<{
  modelValue: boolean
  title: string
}>()

const emits = defineEmits<{
  (e: 'close'): void
}>()
</script>

<style lang="scss" scoped>
.v-dialog {
  margin-right: -1px;

  .v-card-title {
    user-select: none;
    background-color: rgba(var(--v-theme-secondary), 0.2);
    border-bottom: 1px solid rgba(100, 100, 100, 0.2);
  }
}
</style>
