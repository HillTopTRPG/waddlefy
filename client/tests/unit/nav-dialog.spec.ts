import NavDialog from '@/components/NavDialog.vue'
import { VueWrapper } from '@vue/test-utils'
import { createSimpleClassTest, factory, SimpleClassTestInfo } from '../common'

interface NavDialogProps {
  modelValue: boolean
  title: string
}

function factoryWrap(props: Partial<NavDialogProps>) {
  const propsWrap: NavDialogProps = {
    modelValue: true,
    title: 'Sample Title',
    ...props
  }
  return factory(NavDialog, propsWrap)
}

describe('NavDialog.vue', (): void => {
  describe('ダイアログの開閉制御の確認', (): void => {
    describe('modelValueがfalseの場合', (): void => {
      let wrapper: VueWrapper
      beforeEach(() => (wrapper = factoryWrap({ modelValue: false })))
      afterEach(() => wrapper.unmount())

      it('スナップショットテスト', () => expect(wrapper.element).toMatchSnapshot())
      it('ダイアログが非表示状態であること', () => expect(wrapper.find('.v-dialog').exists()).toBeFalsy())
    })
    describe('modelValueがtrueの場合', (): void => {
      let wrapper: VueWrapper
      beforeEach(() => (wrapper = factoryWrap({ modelValue: true })))
      afterEach(() => wrapper.unmount())

      it('スナップショットテスト', () => expect(wrapper.element).toMatchSnapshot())
      it('ダイアログが表示状態であること', () => expect(wrapper.find('.v-dialog').exists()).toBeTruthy())
    })
  })

  describe('コンテンツの確認', (): void => {
    let wrapper: VueWrapper
    beforeEach(() => (wrapper = factoryWrap({ modelValue: true })))
    afterEach(() => wrapper.unmount())

    it('テキストの確認', () => expect(wrapper.find('[data-cy=title-text]').text()).toEqual('Sample Title'))
  })

  describe('閉じるボタンを押した際の挙動', (): void => {
    let wrapper: VueWrapper
    beforeEach(() => {
      wrapper = factoryWrap({ modelValue: true })
      wrapper.find('.v-card-title .v-btn').trigger('click')
      wrapper.vm.$nextTick()
    })
    afterEach(() => wrapper.unmount())

    it('closeが発火すること', () => expect(wrapper.emitted('close')).toEqual([[]]))
    it('ダイアログは表示状態のままであること', () => expect(wrapper.find('.v-dialog').exists()).toBeTruthy())
  })

  describe('classの確認', (): void => {
    const patterns: SimpleClassTestInfo[] = [
      { title: 'タイトル領域がflex-boxであること', selector: '.v-card-title', containClass: 'd-flex' },
      {
        title: 'タイトル領域がjustify-space-betweenであること',
        selector: '.v-card-title',
        containClass: 'justify-space-between'
      },
      { title: 'タイトル領域がalign-centerであること', selector: '.v-card-title', containClass: 'align-center' },
      { title: 'タイトル領域がpy-1であること', selector: '.v-card-title', containClass: 'py-1' },
      {
        title: '閉じるボタンのvariantがtextであること',
        selector: '.v-card-title .v-btn',
        containClass: 'v-btn--variant-text'
      },
      {
        title: '閉じるボタンのsizeがsmallであること',
        selector: '.v-card-title .v-btn',
        containClass: 'v-btn--size-small'
      },
      {
        title: '閉じるボタンのアイコンがmdi-closeであること',
        selector: '.v-card-title .v-btn .v-icon',
        containClass: 'mdi-close'
      }
    ]
    patterns.forEach(createSimpleClassTest.bind(null, factoryWrap.bind(null, { modelValue: true })))
  })
})
