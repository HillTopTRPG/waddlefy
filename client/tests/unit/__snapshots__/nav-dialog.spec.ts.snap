// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeikiBtn.vue > ignoreHeik: falseの場合 > スナップショットテスト 1`] = `
<button
  class="v-btn v-btn--elevated v-theme--light bg-info v-btn--density-comfortable v-btn--size-default v-btn--variant-elevated text-caption px-1"
  hide-details="true"
  style="min-width: auto;"
  type="button"
>
  
  <span
    class="v-btn__overlay"
  />
  <span
    class="v-btn__underlay"
  />
  
  <!---->
  <span
    class="v-btn__content"
    data-no-activator=""
  >
    
    
    <i
      aria-hidden="true"
      class="mdi-emoticon-tongue mdi v-icon notranslate v-theme--light v-icon--size-default"
    />
    <span
      class=""
      data-cy="text"
    >
      平気
    </span>
    
    
  </span>
  <!---->
  <!---->
</button>
`;

exports[`HeikiBtn.vue > ignoreHeiki: trueの場合 > スナップショットテスト 1`] = `
<button
  class="v-btn v-btn--elevated v-theme--light bg-grey v-btn--density-comfortable v-btn--size-default v-btn--variant-elevated text-caption px-1"
  hide-details="true"
  style="min-width: auto;"
  type="button"
>
  
  <span
    class="v-btn__overlay"
  />
  <span
    class="v-btn__underlay"
  />
  
  <!---->
  <span
    class="v-btn__content"
    data-no-activator=""
  >
    
    
    <i
      aria-hidden="true"
      class="mdi-skull mdi v-icon notranslate v-theme--light v-icon--size-default"
    />
    <span
      class="text-decoration-line-through"
      data-cy="text"
    >
      平気
    </span>
    
    
  </span>
  <!---->
  <!---->
</button>
`;

exports[`NavDialog.vue > ignoreHeik: falseの場合 > スナップショットテスト 1`] = `
<div
  data-v-app=""
>
  
  <!---->
  <!---->
  
</div>
`;

exports[`NavDialog.vue > ignoreHeiki: trueの場合 > スナップショットテスト 1`] = `
<div
  data-v-app=""
>
  
  <!---->
  <!---->
  
</div>
`;

exports[`NavDialog.vue > ダイアログの開閉制御の確認 > modelValueがfalseの場合 > スナップショットテスト 1`] = `
<div
  data-v-app=""
>
  
  <!---->
  <!---->
  
</div>
`;

exports[`NavDialog.vue > ダイアログの開閉制御の確認 > modelValueがtrueの場合 > スナップショットテスト 1`] = `
<div
  data-v-app=""
>
  
  <!---->
  <!--teleport start-->
  <div
    aria-modal="true"
    class="v-overlay v-overlay--absolute v-overlay--active v-overlay--contained v-theme--light v-locale--is-ltr v-dialog v-dialog--fullscreen my-16 nav-dialog"
    data-v-d9f6f61e=""
    role="dialog"
    style="z-index: 2400;"
  >
    <transition-stub
      appear="true"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <!---->
    </transition-stub>
    <transition-stub
      appear="true"
      css="true"
      name="slide-y-reverse-transition"
      persisted="true"
    >
      <div
        class="v-overlay__content"
      >
        
        
        
        <div
          class="v-card v-theme--light v-card--density-default v-card--variant-elevated"
          data-v-d9f6f61e=""
        >
          <!---->
          <div
            class="v-card__loader"
          >
            <div
              aria-hidden="true"
              aria-valuemax="100"
              aria-valuemin="0"
              class="v-progress-linear v-theme--light v-locale--is-ltr"
              role="progressbar"
              style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);"
            >
              <!---->
              <div
                class="v-progress-linear__background"
                style="width: 100%;"
              />
              <transition-stub
                appear="false"
                css="true"
                name="fade-transition"
                persisted="false"
              >
                <div
                  class="v-progress-linear__indeterminate"
                >
                  
                  <div
                    class="v-progress-linear__indeterminate long"
                  />
                  <div
                    class="v-progress-linear__indeterminate short"
                  />
                  
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <!---->
          
          <div
            class="v-card-title d-flex justify-space-between align-center py-1"
            data-v-d9f6f61e=""
          >
            <span
              data-cy="title-text"
              data-v-d9f6f61e=""
            >
              Sample Title
            </span>
            <button
              class="v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-small v-btn--variant-text"
              data-v-d9f6f61e=""
              style="font-size: 0.8em;"
              type="button"
            >
              
              <span
                class="v-btn__overlay"
              />
              <span
                class="v-btn__underlay"
              />
              
              <!---->
              <span
                class="v-btn__content"
                data-no-activator=""
              >
                <i
                  aria-hidden="true"
                  class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default"
                />
              </span>
              <!---->
              <!---->
            </button>
          </div>
          
          
          
          <!---->
          
          <!---->
          <span
            class="v-card__underlay"
          />
          
        </div>
        
        
        
      </div>
    </transition-stub>
  </div>
  <!--teleport end-->
  
</div>
`;
